HpToolsLauncher/Launcher.cs

```diff
@@ -121,19 +121,24 @@ public McConnectionInfo()
         public override string ToString()
         {
             string usernameOrClientId;
-            if (!string.IsNullOrEmpty(MobileClientId) && !string.IsNullOrEmpty(MobileSecretKey))
+            if (MobileAuthType == AuthType.AuthToken)
             {
                 usernameOrClientId = string.Format("ClientId: {0}", MobileClientId);
             }
             else

```

Comment:
suggest to use `else if (MobileAuthType == AuthType.UsernamePassword)`, and in `else` block return error message